drivers/flash/spi_nor.c

```diff
@@ -1254,12 +1279,37 @@ static int spi_nor_configure(const struct device *dev)
  */
 static int spi_nor_init(const struct device *dev)
 {
+#if (ANY_INST_HAS_WP_GPIOS || ANY_INST_HAS_HOLD_GPIOS)
+	const struct spi_nor_config *cfg = dev->config;
```

Comment:
Maybe we should have some macro like SPI_NOR_CONFIG or under some other name, that would just return us the cast pointer to `const struct spi_nod_config` made of `dev->config`. I do not like so many ifdefs just to define variable.